lxd/db/warnings.go

```diff
@@ -51,6 +51,31 @@ SELECT warnings.id FROM warnings LEFT JOIN nodes ON warnings.node_id = nodes.id
   WHERE warnings.uuid = ?
 `)
 
+// UpsertWarningLocalNode creates or updates a warning for the local node. Returns error if no local node name.
+func (c *Cluster) UpsertWarningLocalNode(projectName string, entityTypeCode int, entityID int, typeCode WarningType, message string) error {
+	var err error
+	var localName string
+
+	err = c.Transaction(func(tx *ClusterTx) error {
+		localName, err = tx.GetLocalNodeName()
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Failed getting local member name")
+	}
+
+	if localName == "" {
+		return fmt.Errorf("Local member name not available")
+	}
```

Comment:
@tomponline What's the reason we fail here? When running non-clustered LXD the node name is empty and shouldn't fail here. Or am I missing something?

Reply:
In a standalone lxd the local node name is 'none' so if its empty something has gone wrong, which is what I saw during my unrelated cluster upgrade tests that caused the heartbeat request to be unnecessarily ended due to not being able to find the node name due to the cluster not being fully started.

Reply:
As the warnings themselves are stored using the node id we could go further still and remove the indirection and modify the functions to take nodr id directly instead which would also reduce the amount of queries.

Reply:
You're right, it's "none", not empty. I missread my db output. Sorry for the noise ;)